services/console-proxy/server/src/com/cloud/consoleproxy/NoVncConsoleHandler.java

```diff
@@ -0,0 +1,105 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+package com.cloud.consoleproxy;
+
+import com.cloud.consoleproxy.util.Logger;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.handler.AbstractHandler;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.OutputStream;
+
+public class NoVncConsoleHandler extends AbstractHandler {
+    private static final Logger s_logger = Logger.getLogger(NoVncConsoleHandler.class);
+
+    public NoVncConsoleHandler() {
+
+    }
+
+    private void doHandle(Request request, HttpServletResponse httpServletResponse) throws IOException {
+        String queries = request.getUri().getQuery();
+        if (s_logger.isTraceEnabled())
+            s_logger.trace("Handle WebSocket Console request " + queries);
+
+
+        /* no authentication done here
+        *  authentication for the request would be added on websocket connection
+        *  initialization
+        */
+        String[] content =
+                new String[]{"<html>",
+                        "<head>",
+                        "<script type=\"text/javascript\" language=\"javascript\" src=\"/resource/js/jquery.js\"></script>",
+                        "<script> \n" +
+                                "    $(function(){\n" +
+                                "      $(\"#vncpage\").load(\"/resource/noVNC/novnc.html\"); \n" +
+                                "    });\n" +
+                                "    </script>",
+                        " <body> \n" +
+                                "     <div id=\"vncpage\"></div>\n" +
+                                "  </body> ",
+                        "</head>",
+                        "</html>"};
+
+        StringBuffer sb = new StringBuffer();
+        for (int i = 0; i < content.length; i++)
+            sb.append(content[i]);
```

Comment:
Should this string be stored as a static file in `resource` and to consume you can load this file as a resource (from the jar)? In this code, you're converting the array to a stringbuffer to a string, this can be directly change to (1) read the file, (2) use the string.